spec/controllers/users_controller_spec.rb

```diff
@@ -28,4 +32,51 @@
       get :download_pdf, id: 1                            
     end
   end
-end
\ No newline at end of file
+
+  describe "GET /users/:id/edit" do
+    it "should render the edit view for an admin user" do
+      sign_in admin
+      get :edit, id: tenant
+
+      expect(response.status).to eq(200)
+      expect(response).to render_template("edit")
+    end
+
+    it "should redirect a non-admin user to the root path" do
+      sign_in tenant
+      get :edit, id: tenant
+
+      expect(response).to redirect_to(root_path)
+
+      sign_in lawyer
+      get :edit, id: tenant
+
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe "GET /users/:id/update" do
+    let(:params) { { id: tenant, user: { first_name: "Updated" } } }
+
+    it "should update the resource if the request comes from an admin user" do
+      admin.collaborations.create(collaborator: tenant)
```

Comment:
Okay this is actually a safe assumption. You can only get to the edit page for another user if you are a collaborator of theirs. Therefore there will be at least one collaboration: yours.